### PR TITLE
Update `TextSelectionTheme`, `ThemeData`, `TimePicker`, and `TimePickerTheme` tests for M2/M3

### PR DESCRIPTION
**Pull request by QuncCccccc**
Updated unit tests for `TextSelectionTheme`, `ThemeData`, `TimePicker` and `TimePickerTheme` to have M2 and M3 versions.

More info in #127064

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by TahaTesser — APPROVED** [review]
LGTM

File: packages/flutter/test/material/theme_test.dart [hunk 1]
```diff
@@ -684,21 +695,23 @@ void main() {
...
+    });
 
+    testWidgets('Material3 - Can override properties that are independent of material', (WidgetTester tester) async {
       final CupertinoThemeData themeM3 = await testTheme(tester, ThemeData(
-        cupertinoOverrideTheme: const CupertinoThemeData(
-          // The bar colors ignore all things material except brightness.
-          barBackgroundColor: CupertinoColors.black,
-        ),
-        useMaterial3: true
+          cupertinoOverrideTheme: const CupertinoThemeData(
```
Comment (HansMuller) [inline packages/flutter/test/material/theme_test.dart:702]: Looks like the indentation got messed up here - extra spaces were added.

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by TahaTesser** [comment]
I'm waiting for this to land so I can write and update time picker tests on top of this change for https://github.com/flutter/flutter/issues/118657

**Pull request merged**